cpp/src/transform/row_bit_count.cu

```diff
@@ -0,0 +1,508 @@
+                  int parent_index)
+  {
+    out.push_back(col);
+    info.push_back({cur_depth, cur_branch_depth, cur_branch_depth});
+
+    h_info.simple_per_row_size += col.nullable() ? 1 : 0;
+
+    structs_column_view scv(col);
+    auto iter = cudf::detail::make_counting_transform_iterator(
+      0, [&scv](auto i) { return scv.get_sliced_child(i); });
+    flatten_hierarchy(iter,
+                      iter + scv.num_children(),
+                      out,
+                      info,
+                      h_info,
+                      stream,
+                      cur_depth + 1,
+                      cur_branch_depth,
+                      out.size() - 1);
+  }
+
+  // everything else
+  template <typename T,
+            std::enable_if_t<!cudf::is_fixed_width<T>() && !std::is_same<T, string_view>::value &&
+                             !std::is_same<T, list_view>::value &&
+                             !std::is_same<T, struct_view>::value>* = nullptr>
+  void operator()(column_view const& col,
+                  std::vector<cudf::column_view>& out,
+                  std::vector<column_info>& info,
+                  hierarchy_info& h_info,
+                  rmm::cuda_stream_view stream,
+                  size_type cur_depth,
+                  size_type cur_branch_depth,
+                  int parent_index)
+  {
+    CUDF_FAIL("Unsupported column type in row_bit_count");
+  }
+};
+
+template <typename ColIter>
+void flatten_hierarchy(ColIter begin,
+                       ColIter end,
+                       std::vector<cudf::column_view>& out,
+                       std::vector<column_info>& info,
+                       hierarchy_info& h_info,
+                       rmm::cuda_stream_view stream,
+                       size_type cur_depth,
+                       size_type cur_branch_depth,
+                       int parent_index)
+{
+  std::for_each(begin, end, [&](column_view const& col) {
+    cudf::type_dispatcher(col.type(),
+                          flatten_functor{stream},
+                          col,
+                          out,
+                          info,
+                          h_info,
+                          stream,
+                          cur_depth,
+                          cur_branch_depth,
+                          parent_index);
+  });
+}
+
+/**
+ * @brief Struct representing a span of rows.
+ *
+ */
+struct row_span {
+  size_type row_start, row_end;
+};
+
+/**
+ * @brief Functor for computing the size, in bits, of a `row_span` of rows for a given
+ * `column_device_view`
+ *
+ */
+struct row_size_functor {
+  template <typename T>
+  __device__ size_type operator()(column_device_view const& col, row_span const& span)
+  {
+    auto const num_rows{span.row_end - span.row_start};
+    auto const element_size  = sizeof(device_storage_type_t<T>) * 8;
+    auto const validity_size = col.nullable() ? 1 : 0;
+    return (element_size + validity_size) * num_rows;
+  }
+};
+
+template <>
```

Comment:
Could you put the formula in a doxygen comment here?
```
  (rows * offsets + chars) * 8 + (validity-bits)
```
I guess validity-bits is `rows` too. Something like that.